### PR TITLE
[ez] Fix CI log file piping error

### LINKED ISSUE #118764: Print logs on failure in CI
Special case that i need to investigate

Should print logs on failure in CI

ex https://github.com/pytorch/pytorch/actions/runs/7727980439/job/21069062804#step:18:11
https://github.com/pytorch/pytorch/actions/runs/7720187794/job/21050752623#step:18:9

cc @ZainRizvi @kit1980 @huydhn

### PR DESCRIPTION
**Pull request by clee2000**
Fixes https://github.com/pytorch/pytorch/issues/118764

Example log https://github.com/pytorch/pytorch/actions/runs/7737363970/job/21097159160

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
Good catch!

File: test/run_test.py [hunk 1]
```diff
@@ -592,10 +592,16 @@ def print_to_file(s):
...
             REPO_ROOT / ".pytest_cache/v/cache/stepcurrent" / stepcurrent_key
-        ) as f:
+        )
+        if not stepcurrent_file.exists():
+            # Pytest didn't even get to run, this is probably a failure related
+            # to imports in the test file
+            break
+
+        # Read what just failed
+        with open(stepcurrent_file) as f:
```
Comment (malfet) [inline test/run_test.py:604]: Looks fine, but a more Pythonic(and atomic) way would be just to catch exception (as potentially something can delete the file between lines 598  and 604 )

**Review by malfet — APPROVED** [review]
LGTM, but consider:
```
try:
   with open(..) as f:
     ...
catch FileNotFound:
    break
```
as more Pythonic (and less error prone) solution to the problem

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**